### PR TITLE
Initial packaging of Patchwork

### PR DESCRIPTION
**Pull request by ptomato**
Patchwork is a messaging / social app using the Scuttlebutt protocol, which is decentralized, asynchronous, and supposedly offline-friendly.

### CODE REVIEW

File: nz.scuttlebutt.Patchwork.json [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    "command": "run.sh",
+    "separate-locales": false,
+    "finish-args": [
+        "--device=dri",
+        "--share=ipc",
+        "--share=network",
+        "--socket=pulseaudio",
+        "--socket=wayland",
+        "--socket=x11",
+        "--filesystem=~/.ssb:create",
```
Comment (barthalion) [inline nz.scuttlebutt.Patchwork.json]: Use `--persist` instead (also reporting upstream about XDG_CONFIG_DIR would be good).

File: nz.scuttlebutt.Patchwork.json [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                "/app/lib/node_modules/npm/changelogs",
+                "/app/lib/node_modules/npm/doc",
+                "/app/lib/node_modules/npm/html",
+                "/app/lib/node_modules/npm/man",
+                "/app/lib/node_modules/npm/scripts"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://nodejs.org/dist/v8.11.2/node-v8.11.2.tar.xz",
```
Comment (barthalion) [inline nz.scuttlebutt.Patchwork.json]: Does it work with 10.x?

File: nz.scuttlebutt.Patchwork.json [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                {
+                    "type": "git",
+                    "url": "https://github.com/ssbc/patchwork.git",
+                    "tag": "v3.9.0",
+                    "commit": "05199c5427b6df72e01ba59a2d57db4d6d610c2b",
+                    "dest": "main"
+                },
+                {
+                    "type": "file",
+                    "only-arches": ["x86_64"],
```
Comment (barthalion) [inline nz.scuttlebutt.Patchwork.json]: If it works only on x86_64, you also need to add `flathub.json` file with `{"only-arches": ["x86_64"]}`.

**Comment by barthalion** [comment]
bot, build nz.scuttlebutt.Patchwork

File: nz.scuttlebutt.Patchwork.json (hunk 3 above)
Comment (ptomato) [reply nz.scuttlebutt.Patchwork.json]: It _should_ work on other architectures, but these prebuilds were only available for x86_64. I see the build failed without the prebuilds, so I guess I'll try harder to get it to work...

**Comment by barthalion** [comment]
Can't find it on buildbot list so let's try again.

bot, build nz.scuttlebutt.Patchwork

**Comment by barthalion** [comment]
bot, build nz.scuttlebutt.Patchwork

**Comment by barthalion** [comment]
bot, build nz.scuttlebutt.Patchwork

**Comment by barthalion** [comment]
Something is wrong with buildbot, pinged ramcq on IRC.

**Comment by barthalion** [comment]
bot, build nz.scuttlebutt.Patchwork

File: nz.scuttlebutt.Patchwork.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      A decentralized messaging and sharing app built on top of Secure
+      Scuttlebutt (SSB).
+    </p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image type="source">https://raw.githubusercontent.com/ssbc/patchwork/master/screenshot.jpg</image>
+    </screenshot>
+  </screenshots>
+  <categories>
```
Comment (TingPing) [inline nz.scuttlebutt.Patchwork.appdata.xml]: These are in the desktop file so don't need to be duplicated here.

**Comment by ptomato** [comment]
OK, removed the categories and removed the comment from flathub.json.

**Comment by TingPing** [comment]
bot, build nz.scuttlebutt.Patchwork

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/nz.scuttlebutt.Patchwork

**Pull request closed without merging**